rclcpp/test/utils/gtest_macros.hpp

```diff
@@ -0,0 +1,195 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef UTILS__GTEST_MACROS_HPP_
+#define UTILS__GTEST_MACROS_HPP_
+
+#include "gtest/gtest.h"
+
+#include <cstring>
+#include <type_traits>
+
+#include "rclcpp/exceptions/exceptions.hpp"
+
+namespace rclcpp
+{
+namespace testing
+{
+namespace details
+{
+
+/**
+ * \brief Check if two thrown objects are equals.
+ *
+ * For generic thrown objects, probably is unlikely to be used. This type must
```

Comment:
Just out of curiousity, why do you think this is going to be unused?

Reply:
As far as I can tell, all code in rclcpp code either throws a std::exception or an rclcpp::exception type. Seems very unlikely that someone would throw something that doesn't derive from either.